Batch JUnit Test Cases in 2.2

priyankanavale opened the pull request:
JUnit Test Cases for checking batch operations. Checking the CRUD functionality using batch.

Commit:
Batch JUnit Test Cases in 2.2

Landmark-LGC commented on modules/org.restlet.test/src/org/restlet/test/batch/crud/CafeCrudApplication.java:
```diff
@@ -0,0 +1,233 @@
+@SuppressWarnings("unused")
+public class CafeCrudApplication extends Application {
+
+	private static class MyClapRestlet extends Restlet {
```
Rename the class to something that denotes the functionality that you are testing; maybe BatchTestRestlet

Landmark-LGC commented on modules/org.restlet.test/src/org/restlet/test/batch/crud/CafeCrudApplication.java:
```diff
@@ -0,0 +1,233 @@
+		private static final Logger LOGGER = Logger.getLogger(RestletBatchCafeTestCase.class.getName());
+		boolean updatable;
+
+		public MyClapRestlet(Context context, String file, boolean updatable) {
```
Please add comments to all the methods regarding the functionality that they provide. If you are over-riding a method, provide a reference to the parent method.

Landmark-LGC commented on modules/org.restlet.test/src/org/restlet/test/batch/crud/Cafe.java:
```diff
@@ -0,0 +1,118 @@
+
+
+/**
+* Generated by the generator tool for the WCF Data Services extension for the Restlet framework.<br>
```
Change the comments to remove reference to WCF and replace with OData.

Landmark-LGC commented on modules/org.restlet.test/src/org/restlet/test/batch/crud/RestletBatchCafeTestCase.java:
```diff
@@ -0,0 +1,235 @@
+		cafe.setZipCode(cafeZipCode);
+		try {
+
+			BatchRequest br = service.createBatchRequest();
```
Since you are testing the operations separately, please have separate test cases for each of the scenarios here.

Landmark-LGC commented on modules/org.restlet.test/src/org/restlet/test/batch/crud/RestletBatchCafeTestCase.java:
```diff
@@ -0,0 +1,235 @@
+				
+				Query<Cafe> updatequery = service.updateCafeQuery("/Cafes('40')");
+				Cafe cafe3=updatequery.iterator().next();			
+				assertEquals(cafeNameUpdated, cafe3.getName());
```
I don't think the update statement would return an entity; can you check this again?

Landmark-LGC commented:
Covered test cases:
1. Changeset
   a. Create
   b. Update
   c. Delete
2. Get Operation

Additional test cases required:
1. Changeset with more than 1 operation
2. Batch operation with more than 1 changeset
   1. All changesets pass
   2. One changeset fails
3. Batch operation with changeset and get request

Commit:
Merge remote-tracking branch 'upstream/2.2' into 2.2

Commit:
TFS ID: 12967:
RESTLET: Add support for batch operations on the client.
Removed the cast as its not needed with 2.2 version.

Commit:
TFS ID: 12967:
RESTLET: Add support for batch operations on the client.
Junit test cases for batch implementation.

Commit:
TFS ID: 12967:
RESTLET: Add support for batch operations on the client.
Junit test cases for batch implementation.

priyankanavale commented:
Made the respective changes:-
Additional test cases required:
      Changeset with more than 1 operation
Batch operation with more than 1 changeset
      All changesets pass
      One changeset fails
Batch operation with changeset and get request
